### ISSUE TITLE
get_Partition_Count may not be all that useful on Solaris and Derivatives using ZFS

### ISSUE THREAD

**Issue by szaydel** [Seagate/opensea-transport#10]
### Problem

This function pays attention to the` /etc/mnttab` file, looking for information about mounted partitions, but quite commonly, on systems where ZFS is the dominant filesystem, which is likely to be majority of Solaris, illumos and derivatives, there won't be anything referring to individual drives or partitions in `mnttab`, instead there will be ZFS mountpoints. Indeed a given drive may be in a pool and in use, but won't be seen in `mnttab`.

### Expected behavior

I think, if this functionality is to exist on Solaris and friends, it might be necessary to enhance it with checks for ZFS labels. I am not sure about the best approach here, haven't had time to think about. Just wanted to raise this in case it was not already considered.

### How to reproduce

NA

### Deployment information

Anything with ZFS, which at this point would include Linux, BSD, Solaris, illumos, etc.

**Comment by szaydel** [comment]
I guess one of the assumptions being made is that the drive being operated on is not in use. :) I am not sure that there will be any issues like you described, though I suppose we cannot be absolutely sure. Thanks for all the context, this helps. More generally, detecting if a drive has a filesystem, or a part of a filesystem like ZFS would be valuable, I think.

**Comment by szaydel** [comment]
I was thinking a bit more about this, and maybe it is worthwhile making a tiny tweak to what is output by the programs, so that instead of `Partition count for` it is `Active partition count for`, or some other word to replace active. It seems worthwhile to clarify that this is not an actual partition count because it might be misleading if you believe that there is at least one or more partitions on the device.

**Comment by vonericsen** [comment]
Some operations should not be run while a partition is mounted, and others it is ok. It depends a lot on what is being done.

I like the idea of something like "active" or maybe "detected". I will think about the wording a bit to see if there is a better way to describe it to make sure it informs without misleading anyone.

**Comment by szaydel** [comment]
Yeah, terminology here is key. Right now, I think language is quite misleading and not really helpful. Thanks for giving this a thought.

**Comment by vonericsen** [comment]
I have been continuing to think about this issue to figure out a solution.
I pushed a change to rename the variable to be slightly more clear. I decided that "active" made a lot of sense to use instead.

For ZFS, I did some reading on how it works, config files, and the various zfs and zpool commands to get an idea of what we can do.
It looks like the file `/etc/zfs/zpool.cache` may be able to be parsed to check for drives containing a zfs pool. I will need to do more research before we try doing this as well as figuring out how to properly parse this file, but I think this could be a solution for systems using ZFS file systems.

**Comment by vonericsen** [comment]
I will look into the label as well to see how that detection could be added!
Thanks for this idea too!
